### PR TITLE
Add blog post on how Android builds work in Bazel

### PR DESCRIPTION
**Pull request by asteinb**

### CODE REVIEW

File: _posts/2018-02-14-how-android-builds-work-in-bazel.md [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+rule has access to its direct dependencies’ providers. Because most information
+passed between rules is actually transitive, providers make use of the
+[`NestedSet`](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/collect/nestedset/NestedSet.java)
+class, a DAG-like data structure (it’s not actually a set!) made up of items and
+pointers to other (nested) `NestedSet` objects. `NestedSet`s are specially
+optimized to work efficiently for analysis. For part of a provider that
+represents some transitive state, for example, a trivial implementation might be
+to build a new list that contains the items for the current rule and each
+transitive dependency (for a chain of n transitive dependencies, that means we’d
+add $$n + (n - 1) + … 1 = O(n^2)$$ items to some list), but building a nested
```
Comment (jin) [inline _posts/2018-02-14-how-android-builds-work-in-bazel.md]: I don't think we do any conversion of math syntax, remove the `$$`?

File: _posts/2018-02-14-how-android-builds-work-in-bazel.md [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+set containing the new item and a pointer to the previous nested set is much
+more efficient (we’d add $$2 + 2 + … 2 = O(n)$$ items to some nested set). This
```
Comment (jin) [inline _posts/2018-02-14-how-android-builds-work-in-bazel.md]: rm math syntax delimiter here

File: _posts/2018-02-14-how-android-builds-work-in-bazel.md (hunk 1 above)
Comment (asteinb) [reply _posts/2018-02-14-how-android-builds-work-in-bazel.md]: removed

File: _posts/2018-02-14-how-android-builds-work-in-bazel.md (hunk 2 above)
Comment (asteinb) [reply _posts/2018-02-14-how-android-builds-work-in-bazel.md]: removed

**Comment by jin** [comment]
Thanks!

**Pull request merged**